build.gradle

```diff
@@ -1,32 +1,68 @@
-buildscript {
-    repositories {
-        gradlePluginPortal()
-        maven {
-            url = "https://s01.oss.sonatype.org/content/repositories/snapshots/"
-        }
-    }
-    dependencies {
-        classpath "com.exactpro.th2:th2-gradle-plugin:0.0.1-dev-5915772757-13a28ae-SNAPSHOT"
-    }
+plugins {
+    id "org.jetbrains.kotlin.jvm" version "1.8.22"
+    id "org.jetbrains.kotlin.kapt" version "1.8.22"
+    id("java-library")
+    id("maven-publish")
+
+    id "com.exactpro.th2.gradle.publish" version "0.0.6"
+    id "com.exactpro.th2.gradle.component" version "0.0.6"
 }
 
-apply plugin: "com.exactpro.th2.common-conventions"
-apply plugin: "com.exactpro.th2.docker-conventions"
+group = 'com.exactpro.th2'
+version = release_version
+
+kotlin {
+    jvmToolchain(11)
+}
 
-apply plugin: 'kotlin-kapt'
+repositories {
+    mavenCentral()
+    maven {
+        name 'Sonatype_snapshots'
+        url 'https://s01.oss.sonatype.org/content/repositories/snapshots/'
+    }
+    maven {
+        name 'Sonatype_releases'
+        url 'https://s01.oss.sonatype.org/content/repositories/releases/'
+    }
+    mavenLocal()
+
+    configurations.configureEach {
+        resolutionStrategy.cacheChangingModulesFor 0, 'seconds'
+        resolutionStrategy.cacheDynamicVersionsFor 0, 'seconds'
+    }
+}
 
 dependencies {
-    api platform("com.exactpro.th2:bom:4.5.0")
-    implementation "com.exactpro.th2:common:5.7.2-dev"
-    implementation "com.exactpro.th2:common-utils:2.2.2-dev"
-    implementation "com.exactpro.th2:codec:5.4.1-dev"
+    implementation "com.exactpro.th2:common:5.10.1-dev"
+    implementation "com.exactpro.th2:common-utils:2.2.3-dev"
+    implementation "com.exactpro.th2:codec:5.5.0-th2-gradle-plugin-+"
 
     implementation "org.apache.commons:commons-lang3"
     implementation "com.fasterxml.jackson.core:jackson-databind"
 
     implementation "net.sourceforge.javacsv:javacsv:2.0"
 
+    implementation "io.github.microutils:kotlin-logging:3.0.5"
+
     compileOnly "com.google.auto.service:auto-service-annotations:1.1.1"
-    annotationProcessor "com.google.auto.service:auto-service:1.1.1"
```

Comment:
Why did you remove it? The codec factory is located in Java sources not in Kotlin sources. I am not sure that kapt will process java source. Could you please check that the corresponding service files are generated?